enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -749,29 +749,20 @@ func (p *PebbleCache) migrateData(quitChan chan struct{}) error {
 }
 
 func (p *PebbleCache) processAccessTimeUpdates(quitChan chan struct{}) error {
-	exitMu := sync.Mutex{}
-	exiting := false
-	go func() {
-		<-quitChan
-		exitMu.Lock()
-		exiting = true
-		exitMu.Unlock()
-	}()
-
 	for {
 		select {
 		case accessTimeUpdate := <-p.accesses:
 			if err := p.updateAtime(accessTimeUpdate.key); err != nil {
 				log.Warningf("Error updating atime: %s", err)
 			}
-		default:
-			exitMu.Lock()
-			done := exiting
-			exitMu.Unlock()
-
-			if done {
-				return nil
+		case <-quitChan:
+			// Drain any updates in the queue before exiting.
+			for u := range p.accesses {
```

Comment:
does `p.accesses` ever get closed?

Reply:
after a quick read through it looks like `p.accesses` doesn't get closed so I think this will hang until the server is forcibly shut down

Reply:
it doesn't bleh

Reply:
fixed